actionpack/lib/action_dispatch/routing/redirection.rb

```diff
@@ -24,7 +24,11 @@ def call(env)
       def serve(req)
         req.check_path_parameters!
         uri = URI.parse(path(req.path_parameters, req))
-        
+
+        if uri.opaque
+          raise ArgumentError, "redirect does not support opaque targets"
```

Comment:
Should we include `uri` in the error message to make it easier to figure out what's going on?